binary-search-tree/ruby/inventory_management.rb

```diff
@@ -34,17 +34,36 @@ def initialize
 
   def search(id)
     # Comment out the line below and make changes here
-    raise "you must implement search"
+    # raise "you must implement search"
```

Comment:
It might help to define 2 separate methods `search_iterative` (which represents your initial implementation) and `search_recursive` which represents. Then inside `search` you can decide the one you want to call e.g.

```ruby
def search(id)
   search_iterative(id)
end
```